.github/workflows/pr.yml

```diff
@@ -70,10 +70,16 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-go-
     - name: install deps
-      run: go install golang.org/x/tools/cmd/goimports@latest && go install github.com/klauspost/asmfmt/cmd/asmfmt@latest
+      run: |
+        go install golang.org/x/tools/cmd/goimports@latest && go install github.com/klauspost/asmfmt/cmd/asmfmt@latest
+        go install github.com/ethereum/go-ethereum/cmd/abigen@v1.12.0
+        go install github.com/consensys/gnark-solidity-checker@latest
+        sudo add-apt-repository ppa:ethereum/ethereum
+        sudo apt-get update
+        sudo apt-get install solc
     - name: Test
       run: |
-        go test -v -short -timeout=30m ./...
+        go test -v -short -tags=solccheck -timeout=30m ./...
```

Comment:
The Solitify check seems quite heavy - maybe it would be worth to create separate workflow and call less frequently? I think this could slow down the development speed quite a lot, considering Solidity tests are run for the whole integration test suite.

Reply:
Hmm, I guess we can include solidity tests in CI for now and then later figure out how to make PR tests lighter (I think it would be good to get faster feedback when developing).